<div style='background:black; padding:5px; border-radius:5px'>
    <h1 style='text-align:center;color:#BD2A2E'>Medical Appointment No Shows </h1>
    <hr style="width:80%; color:blue">
    <hr style="width:50%; color:blue">
    <img src='../src/image/01.jpeg' alt='project image' style='width:500px; height:300px; display: block; margin: auto;'>
    <h3 style='color:white'>Project Description : </h3>
    <p style='color:white;text-align:center'>This project aims to analyze a dataset of 100,000 medical appointments in Brazil, focusing on whether or not patients show up for their appointments. The dataset includes several characteristics about the patient in each row.
    </p>
    <h3>Data : </h3>
    <table style="width:80%;margin: auto;">
    <thead>
       <tr>
            <th style="border:1px solid black;padding:8px;width:30%;text-align:center;color: #BD2A2E">Column</th>
            <th style="border:1px solid black;padding:8px;text-align:center;color: #BD2A2E">Description</th>
        </tr>
    </thead>
    <tbody>
        <tr>
            <td style="text-align:center">PatientId</td>
            <td>Identification of a patient.</td>
        </tr>
        <tr>
            <td style="text-align:center">AppointmentID</td>
            <td>Identification of each appointment.</td>
        </tr>
        <tr>
            <td style="text-align:center">Gender</td>
            <td> Male or Female. Female is the greater proportion, women take way more care of their health in comparison to men</td>
        </tr>    
        <tr>
            <td style="text-align:center">DataMarcacaoConsulta</td>
            <td> The day of the actual appointment, when they have to visit the doctor.</td>
        </tr>   
        <tr>
            <td style="text-align:center">DataAgendamento</td>
            <td> The day someone called or registered the appointment, this is before the appointment of course.</td>
        </tr>   
        <tr>
            <td style="text-align:center">Age</td>
            <td> How old is the patient.</td>
        </tr>     
        <tr>
            <td style="text-align:center">Neighbourhood</td>
            <td> Where the appointment takes place.</td>
        </tr>  
        <tr>
            <td style="text-align:center">Scholarship</td>
            <td> Indicates whether or not the patient is enrolled in  welfare program</td>
        </tr>     
         <tr>
            <td style="text-align:center">Hipertension</td>
            <td>  Indicates if the patient has hypertension</td>
        </tr>
        <tr>
            <td style="text-align:center">Diabetes</td>
            <td>  Indicates if the patient has diabetes.</td>
        </tr>
         <tr>
            <td style="text-align:center">Alcoholism</td>
            <td>  Indicates if the patient is an alcoholic.</td>
        </tr>
        <tr>
            <td style="text-align:center">Handcap</td>
            <td> The number of disabilities a patient has</td>
        </tr>
        <tr>
            <td style="text-align:center">SMS_received</td>
            <td> Indicates if one or more SMS messages were sent to the patient. This is an integer</td>
        </tr>
        <tr>
            <td style="text-align:center">No-show</td>
            <td>  It says ‘No’ if the patient showed up to their appointment, and ‘Yes’ if they did not show up.</td>
        </tr>
    </tbody>
    </table>
</div>


<hr style="width:95%">
<hr style="width:70%">
<hr style="width:50%">
<hr style="width:25%">
<hr style="width:10%">
<hr style="width:5%">

<div style="background:#4C4A59;width:900px;margin:auto;border-radius:8px;padding:8px">
    <h1 style="color:#DBF227;text-align:center;">Import Libraries</h1>
</div>

In [40]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
import plotly_express as px
import klib

In [22]:
#load the data
df=pd.read_csv('../data/external/KaggleV2-May-2016.csv')

In [23]:
df.shape

(110527, 14)

In [24]:
#the name of columns
df.columns

Index(['PatientId', 'AppointmentID', 'Gender', 'ScheduledDay',
       'AppointmentDay', 'Age', 'Neighbourhood', 'Scholarship', 'Hipertension',
       'Diabetes', 'Alcoholism', 'Handcap', 'SMS_received', 'No-show'],
      dtype='object')

In [20]:
#show the first 5 rows of the dataset
df.head()

patient_id  appointmentid gender         scheduled_day  \
0  2.987250e+13        5642903      F  2016-04-29T18:38:08Z   
1  5.589978e+14        5642503      M  2016-04-29T16:08:27Z   
2  4.262962e+12        5642549      F  2016-04-29T16:19:04Z   
3  8.679512e+11        5642828      F  2016-04-29T17:29:31Z   
4  8.841186e+12        5642494      F  2016-04-29T16:07:23Z   

        appointment_day  age      neighbourhood  scholarship  hipertension  \
0  2016-04-29T00:00:00Z   62    JARDIM DA PENHA            0             1   
1  2016-04-29T00:00:00Z   56    JARDIM DA PENHA            0             0   
2  2016-04-29T00:00:00Z   62      MATA DA PRAIA            0             0   
3  2016-04-29T00:00:00Z    8  PONTAL DE CAMBURI            0             0   
4  2016-04-29T00:00:00Z   56    JARDIM DA PENHA            0             1   

   diabetes  alcoholism  handcap  sms_received no_show  
0         0           0        0             0      No  
1         0           0        0             0      No  
2         0           0        0             0      No  
3         0           0        0             0      No  
4         1           0        0             0      No

<div style="background: #49243E; width: 800px; color: #DBAFA0; border-radius: 10px; margin: auto; padding: 20px;">
    <ul >
        <li>first our data contains of 110527 rows and 14 features </li>
        <li>some columns need to fixed and them make all columns in lower case </li>
    </ul>
</div>

In [25]:
# fixed the name of columns 
df.rename(columns={'AppointmentID':'Appointment_ID','PatientId':'patient_id','ScheduledDay':'Scheduled_Day','AppointmentDay':'Appointment_Day','No-show':'No_show'},inplace=True)

# make the columns in lower case 
df.columns=df.columns.str.lower()

In [26]:
#check the value of each columns 

for col in df.select_dtypes('object_').drop(['scheduled_day','appointment_day'],axis=1).columns:
    display(df[col].value_counts().to_frame().T)

gender      F      M
count   71840  38687

neighbourhood  JARDIM CAMBURI  MARIA ORTIZ  RESISTÊNCIA  JARDIM DA PENHA  \
count                    7717         5805         4431             3877   

neighbourhood  ITARARÉ  CENTRO  TABUAZEIRO  SANTA MARTHA  JESUS DE NAZARETH  \
count             3514    3334        3132          3131               2853   

neighbourhood  BONFIM  ...  UNIVERSITÁRIO  SEGURANÇA DO LAR  NAZARETH  \
count            2773  ...            152               145       135   

neighbourhood  MORADA DE CAMBURI  PONTAL DE CAMBURI  ILHA DO BOI  \
count                         96                 69           35   

neighbourhood  ILHA DO FRADE  AEROPORTO  ILHAS OCEÂNICAS DE TRINDADE  \
count                     10          8                            2   

neighbourhood  PARQUE INDUSTRIAL  
count                          1  

[1 rows x 81 columns]

no_show     No    Yes
count    88208  22319

<div style="background:#527853;width:900px;margin:auto;border-radius:8px">
    <h2 style="color:#F9E8D9;text-align:center;">Info about the data</h2>
</div>

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110527 entries, 0 to 110526
Data columns (total 14 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   patient_id       110527 non-null  float64
 1   appointmentid    110527 non-null  int64  
 2   gender           110527 non-null  object 
 3   scheduled_day    110527 non-null  object 
 4   appointment_day  110527 non-null  object 
 5   age              110527 non-null  int64  
 6   neighbourhood    110527 non-null  object 
 7   scholarship      110527 non-null  int64  
 8   hipertension     110527 non-null  int64  
 9   diabetes         110527 non-null  int64  
 10  alcoholism       110527 non-null  int64  
 11  handcap          110527 non-null  int64  
 12  sms_received     110527 non-null  int64  
 13  no_show          110527 non-null  object 
dtypes: float64(1), int64(8), object(5)
memory usage: 11.8+ MB


<div style="background: #49243E; width: 800px; color: #DBAFA0; border-radius: 10px; margin: auto; padding: 20px;">
    <ul >
        <li>first our target is <b>no_show</b>  ,No if the patient showed up to their appointment, and Yes if they did not show up  </li>
        <li> the data contains 3 categorical columns and 2 date type columns and 8 numerical columns and index columns <b>patient_id</b> </li>
        <li>it seem that the data not have null value </li>
    </ul>
</div>

In [19]:
#describe of the data 
df.describe().T.style.background_gradient()

<div style="background: #49243E; width: 800px; color: #DBAFA0; border-radius: 10px; margin: auto; padding: 20px;">
    <ul >
        <li><b>Age</b> : the average patient age is (37) years and the min value is (-1) that may be can an error   </li>
        <li>  <b>Scholarship</b> on average about 9.8% of patient are enrolled in welfare program   </li>
        <li><b> hypertension </b> : on average about 19.72 of patents have hypertension  </li>
        <li><b> Diabetes </b> : on average about 7.19% of patents have Diabetes  </li>
        <li><b> sms_received </b> : On average, about 32.10% of the patients received one or more SMS messages.  </li>
        <li><b> handcap </b> : on average about 2.22% of patents are disabilities  and the 97% of patents not have disabilities that appears on 25% and 50% and 75% of data have 0 value    </li>
    </ul>
</div>

In [46]:
#drop the row when the age value is -1

df[df['age']==-1]

Empty DataFrame
Columns: [patient_id, appointment_id, gender, scheduled_day, appointment_day, age, neighbourhood, scholarship, hipertension, diabetes, alcoholism, handcap, sms_received, no_show]
Index: []

In [45]:
df=df[df['age']!=-1]

In [36]:
df.describe(include='object').T.style.background_gradient()

<div style="background: #49243E; width: 800px; color: #DBAFA0; border-radius: 10px; margin: auto; padding: 20px;">
    <ul >
        <li><b>Gender</b> : There are 2 unique values in this column. The most frequent gender is Female (F), appearing 71,840 times.</li>
        <li><b>neighbourhood</b> : There are 81 unique values in this column. The most frequent neighbourhood is JARDIM CAMBURI, appearing 7,717 times.</li>
        <li><b>no_show</b>There are 2 unique values in this column. The most frequent value is No, appearing 88,208 times. This means that most appointments were kept.</li>
        <li>the target columns <b> ( no_show)</b> has imbalanced data ,the value of no appear 80% of the data </li>
        </ul>


</div>

In [41]:
#check the null value 
klib.missingval_plot(df)

No missing values found in the dataset.


In [43]:
#check the duplicated in data
df.duplicated().sum()

0

In [48]:
df.to_csv('../data/processed/patent_data.csv')

<div style="background:#97BFB4;width:900px;margin:auto;border-radius:8px;padding:8px">
    <p style="color:#EC255A;text-align:center;"> here we can end this book in this step end this book and move to next step </p>
</div>

<img src='../src/image/02.jpeg' alt='project image' style='width:500px; height:300px; display: block; margin: auto;'>
